presto-iceberg/src/main/java/com/facebook/presto/iceberg/util/StatisticsUtil.java

```diff
@@ -89,7 +89,7 @@ public static TableStatistics mergeHiveStatistics(TableStatistics icebergStatist
             }
             statsBuilder.setColumnStatistics(columnHandle, mergedStats.build());
         });
-        return calculateAndSetTableSize(statsBuilder).build();
+        return calculateAndSetTableSize(statsBuilder).setConfidenceLevel(statsBuilder.build().getConfidence()).build();
```

Comment:
is this right? statsBuilder never sets the confidence, so seems like this will always be low. I think you first need to set it to the min confidence level as you are iterating through.  Also, would be good to add a test for this.

Reply:
@rschlussel only icebergStats have fine grained statistics, so only place where we can take confidenceLevel from is there.